framework/operator.go

```diff
@@ -2,8 +2,55 @@ package framework
 
 import (
 	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
 )
 
+// Config represents the configuration used to create a new operator framework.
+type Config struct {
+	// Dependencies.
+	Logger micrologger.Logger
+}
+
+// DefaultConfig provides a default configuration to create a new operator
+// framework by best effort.
+func DefaultConfig() Config {
+	var err error
+
+	var newLogger micrologger.Logger
+	{
+		config := micrologger.DefaultConfig()
+		newLogger, err = micrologger.New(config)
+		if err != nil {
+			panic(err)
+		}
+	}
```

Comment:
> Lets talk about this separately.

Yes, but why not here :)

Usually `Config` looks like this:

```
type Config struct {

        // Dependencies.

        // Settings.
}
```

I think only settings should have default values. This makes a lot of sense for me. But dependencies should be provided. Otherwise funny things may happen. Like different instance of logger or kubernetes client in single struct instance.

If you want a test logger we use, then let's make a `loggertest` package in `micrologger`. I have some ideas, and can create a PR. It would follow an approach of `httptest` package in go stdlib.